vgen/src/main/java/io/gapi/vgen/csharp/CSharpMethodViewing.java

```diff
@@ -0,0 +1,29 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gapi.vgen.csharp;
+
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.ProtoFile;
+
+import io.gapi.vgen.MethodViewing;
+
+public class CSharpMethodViewing extends MethodViewing
+    implements CSharpProtoElementViewing<Method> {
+
+  @Override
+  public ProtoFile getNamespaceFile(Method method) {
+    return method.getParent().getFile();
```

Comment:
It doesn't seem generic to me right now. Every language seems to have its own custom code for implementing `GapicLanguageProvider.generate()`. I would prefer not to put a method into ProtoElementViewing / InputElementView unless it is used in every subclass.